tests/Basics/BuildIdTest.cpp

```diff
@@ -0,0 +1,34 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Markus Pfeiffer
+////////////////////////////////////////////////////////////////////////////////
+#include <gtest/gtest.h>
+
+#include "Basics/BuildId.h"
+
+using namespace arangodb::build_id;
+
+TEST(BuildIdTest, successfully_obtains_build_id) {
+  if constexpr (arangodb::build_id::supportsBuildId()) {
+    auto buildId = getBuildId();
+    ASSERT_EQ(buildId.size(), 20);
```

Comment:
Note that the build-id size seems to be variable in reality.
When I built arangod locally using clang, I got build-ids that were 8 bytes long (16 bytes in hex), whereas when I built with Oskar, I got build-ids that were 10 bytes long (20 bytes in hex).
The actual length seems to depend on the build-id "style".
From `man ld`:
```
       --build-id
       --build-id=style
           Request the creation of a ".note.gnu.build-id" ELF note section or
           a ".buildid" COFF section.  The contents of the note are unique
           bits identifying this linked file.  style can be "uuid" to use 128
           random bits, "sha1" to use a 160-bit SHA1 hash on the normative
           parts of the output contents, "md5" to use a 128-bit MD5 hash on
           the normative parts of the output contents, or "0xhexstring" to use
           a chosen bit string specified as an even number of hexadecimal
           digits ("-" and ":" characters between digit pairs are ignored).
           If style is omitted, "sha1" is used.
```

Reply:
Fair enough; the 20 was just a potshot to see whether there is *anything* I can test that has meaning.